### DISCUSSION TITLE
[v4] Tailwind breaks after upgrading to 4.0.0-alpha.32 [SOLVED]

### DISCUSSION THREAD

**Discussion by ev3rlost** [tailwindlabs/tailwindcss#14961] [Help]
After upgrading to 4.0.0-alpha.32 most of my projects using the Alpha v4 are now broken.

I get errors such as:

`Cannot apply unknown utility class: px-2`  or `Cannot apply unknown utility class: text-xxs`.

I did rename `--font-size-*` variables to `--text-*` in my `@theme`

Is this a bug, should I downgrade for now, or am I doing something wrong here?

**Comment by adamwathan** [comment]
Hey thanks for reporting! Can you share some code so I can try and reproduce it locally?

**Reply by adamwathan** [reply]
Seems to work here for instance so need to know what the trigger is:

https://play.tailwindcss.com/7lRg8YMgmp?file=css

**Comment by ev3rlost** [comment]
`app.css` file:

```
@import 'tailwindcss';

@import 'fonts.css' layer(base);
@import 'utilities.css' layer(utilities);
@import 'components.css' layer(components);

@plugin '@tailwindcss/forms';

@theme {
    --text-xxs: 0.65rem;
    ...etc
}
```

`components.css` file:

```
@import 'alerts.css';
... etc
```

`alerts.css` file:

```
.alert {
    @apply px-2 py-1.5 rounded-md text-sm select-none;
}
```

I can see it working in your playground though :/

**Reply by adamwathan** [reply]
Can you create a simple repo that reproduces the issue and share it?

**Reply by ev3rlost** [reply]
Before doing that I restarted Chrome and VSCode - and now it's working as intended. Any idea why VSCode would do that? (it does kinda 'freeze' often (instellisense stops working for example) if its open for a longer period of time). I did run npm run dev again though and ran php artisan optimize, but that didn't fix it until I completely closed the editor and restarted it.

I think this was all on my end and has nothing to do with the new alpha release, so this thread can be considered pointless. 

Sorry for the troubles! I thought it had to do with the new release. Anyway, thanks for your time Adam!

**Comment by krishnaTORQUE** [comment]
I am having the same issue.
I ran `npx @tailwindcss/upgrade@next`, then tried to build the application again and got the error below.
```
error during build:
[@tailwindcss/vite:generate:build] Cannot apply unknown utility class: lg:w-1/3
```

**Reply by wongjn** [reply]
You may need to use [`@reference`](https://tailwindcss.com/docs/functions-and-directives#reference-directive) somewhere.

---
As an aside, Adam Wathan (creator of Tailwind) does seem to advocate avoiding `@apply`:
- https://twitter.com/adamwathan/status/1226511611592085504
- https://twitter.com/adamwathan/status/1559250403547652097
